catalog/spatial/registry/registry-rest-endpoint/src/main/java/org/codice/ddf/registry/rest/endpoint/RegistryRestEndpoint.java

```diff
@@ -146,24 +143,150 @@ public Response viewRegInfoHtml(@PathParam("registryId") final String registryId
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(getErrorHtml(message))
                     .build();
         }
 
-        Map<String, Object> registryMap = reportMapBuilder.buildRegistryMap(registryPackage);
+        if (!isValidSection(section)) {
 
+            String message =
+                    String.format("'%s' is an unknown section name. Valid Sections are: %s",
+                            section,
+                            validSections);
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(getErrorHtml(message))
+                    .build();
+        }
+        RegistryPackageType registryPackage;
         try {
-            Handlebars handlebars = new Handlebars(templateLoader);
-            Template template = handlebars.compile("report");
-            html = template.apply(registryMap);
+            Set<String> sourceIdsSet = new HashSet<>();
+            if (CollectionUtils.isNotEmpty(sourceIds)) {
+                sourceIdsSet.addAll(sourceIds);
+            }
+            registryPackage = federationAdminService.getRegistryObjectByRegistryId(registryId,
+                    sourceIdsSet);
+        } catch (FederationAdminException e) {
+            String message = "Error getting registry package.";
+            LOGGER.error("{} For registry id: '{}', optional sources: {}",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(getErrorHtml(message))
+                    .build();
+        }
+
+        if (registryPackage == null) {
+            String message = "No registry package was found.";
+            LOGGER.error("{} For registry id: '{}', optional source ids: {}.",
+                    message,
+                    registryId,
+                    sourceIds);
+            return Response.status(Response.Status.NOT_FOUND)
+                    .entity(getErrorHtml(message))
+                    .build();
+        }
+
+        try {
+            html = registryReportBuilder.getHtmlFromRegistryPackage(registryPackage, section);
         } catch (IOException e) {
-            LOGGER.error("Error compiling and applying report template.");
+            String message = String.format("Error when compiling and applying %s template.",
+                    section);
+            LOGGER.error("{} For registry id: '{}'", message, registryId);
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR)
+                    .entity(getErrorHtml(message))
+                    .build();
         }
 
         return Response.ok(html)
                 .build();
     }
 
+    private String getErrorHtml(String error) {
+        return String.format("<html>%s</html>", error);
```

Comment:
Though this will probably work with most parsers, it is still a long way off from what is needed at a minimum to be valid.

http://www.brucelawson.co.uk/2010/a-minimal-html5-document/
https://www.sitepoint.com/a-minimal-html-document-html5-edition/